### PR TITLE
Fix issue with mapped domains.

### LINKED ISSUE #48: Ads.txt file detection does not take mapped domains into account
**Describe the bug**
If you use a mapped domain (i.e. your wp-admin is on a different domain from your site, like on WordPress.com), the ads.txt file detection looks for the file on the wp-admin domain, which can return a false positive. See https://wordpress.org/support/topic/existing-ads-txt-file-found/

**Steps to Reproduce**
I haven't actually done this yet, but my guess is:

1. Put an ads.txt in the root of a MS instance
2. Set up domain mapping for a subsite
3. Go to the ads.txt edit screen on the subsite
4. See notice that an existing ads.txt file was found

**Expected behavior**
I would not expect to see the notice.

**Screenshots**
n/a

**Environment information**
This is as of Ads.txt 1.2.0, seen on WordPress.com with a business plan where you can install plugins and map domains.

**Additional context**
Reported here: https://wordpress.org/support/topic/existing-ads-txt-file-found/

### PR DESCRIPTION
**Pull request by mmcachran**
### Description of the Change

Fix issue with mapped domains by using the admin URL instead of the FE URL.

### Verification Process

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Applicable Issues

Closes https://github.com/10up/ads-txt/issues/48

### CODE REVIEW

**Review by helen — COMMENTED** [review]
Fixing some Travis issues

**Comment by jeffpaul** [comment]
@mmcachran looks like there's a PHPCS issue, see https://github.com/10up/ads-txt/pull/53/checks?check_run_id=533621057, would you please look at resolving that as you have the time?

**Comment by jeffpaul** [comment]
Pinging @dinhtungdu for review on this.

**Review by dinhtungdu — COMMENTED** [review]
In my test, I can save the ads.txt content in the mapped domain sub-site flawlessly without this fix so I'm not sure which problem we have with the Ajax URL.

Note that we removed the detection warning [here](https://wordpress.org/support/topic/existing-ads-txt-file-found/#post-12824432).

**Comment by jeffpaul** [comment]
@mmcachran are you still seeing this issue occur without this PR applied?

File: js/admin.js [hunk 1]
```diff
@@ -29,7 +29,7 @@
 		$.ajax({
 			type: 'POST',
 			dataType: 'json',
-			url: ajaxurl,
+			url: window.adstxt.ajaxUrl,
```
Comment (peterwilsoncc) [inline js/admin.js:32]: I don't think this is what was intended by the original ticket, as the admin-ajax URL is currently working as expected.

**Review by peterwilsoncc — COMMENTED** [review]
I was asked to take another look at this PR.

As @dinhtungdu highlighted above, the conflict checker was removed in c64d09d850a45a8e9fa58d65d4b4fce92bdf5cd7. Without reverting the commit, this has no affect. 

Rather than coding the URL used for `ajaxUrl` to a custom version, the code needs to be passed the full URLs for the `ads.txt` and `app-ads.txt` file.

Additionally, to ensure the domain is file can be requested from the admin, we'll need to make sure the `Access-Control-Allow-Origin` is set to include both the front-end (mapped) and back-end domains.

**Comment by jeffpaul** [comment]
@peterwilsoncc sounds like we might be best served by closing this PR and handling via an alternate approach/PR?

**Comment by peterwilsoncc** [comment]
@jeffpaul Yes, a different approach will need to be found. I'll close this PR.

I did some further testing last week on the approach in this PR (testing for ads.txt client side) and for sites with domain mapping enabled, doing so causes [CORS Errors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors) as the domain is different.

As the test is for the presence of a real file, the plugin is unable to modify the HTTP headers to allow for the cross origin request. 

Thank you for taking a look at this @mmcachran, it did help with discovering an approach even if the code can't be used directly.

**Comment by peterwilsoncc** [comment]
PR #131 has been merged to resolve this, I've included the folks that helped on this PR in the props list as this was a required step to figure out the best approach.

**Pull request closed without merging**